jaeger-context/src/main/java/com/uber/jaeger/context/ActiveSpanSourceTraceContext.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2017, Uber Technologies, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.uber.jaeger.context;
+
+import io.opentracing.ActiveSpan;
+import io.opentracing.ActiveSpanSource;
+import io.opentracing.Span;
+import io.opentracing.Tracer;
+import io.opentracing.util.GlobalTracer;
+import io.opentracing.util.ThreadLocalActiveSpan;
+
+import java.lang.reflect.Field;
+
+/**
+ * This is a {@link TraceContext} that relies on the {@link ActiveSpanSource} registered with {@link
+ * GlobalTracer}.
+ */
+public class ActiveSpanSourceTraceContext implements TraceContext {
+
+  private final Tracer tracer;
+  /**
+   * This is a hack to retrieve the span wrapped by the {@link ThreadLocalActiveSpan} implementation
+   * to shoehorn into the {@link TraceContext} implementation. This is being done so that
+   * instrumentation relying on {@link Tracer} has a is consistent with instrumentation using
+   * {@link TraceContext}.
+   */
+  private static final Field wrappedSpan;
+
+  static {
+    try {
+      wrappedSpan = ThreadLocalActiveSpan.class.getDeclaredField("wrapped");
+      wrappedSpan.setAccessible(true);
+    } catch (NoSuchFieldException e) {
+      throw new RuntimeException("Unable to access ThreadLocalActiveSpan.wrapped reflectively.", e);
+    }
+  }
+
+  public ActiveSpanSourceTraceContext(Tracer tracer) {
```

Comment:
* I think you can use a more narrow dependency, ActiveSpanSource instead of Tracer.
  * instead of depending on ActiveSpanSource, it makes sense to depend directly on ThreadLocalActiveSpanSource

Reply:
>instead of depending on ActiveSpanSource, it makes sense to depend directly on ThreadLocalActiveSpanSource

This is a bit involved because we need to use reflection to get the instance of the `ThreadLocalActiveSpanSource` from the tracer. I updated to use the `ActiveSpanSource` interface instead.

Reply:
you don't need reflection, just a cast: `(ThreadLocalActiveSpanSource) tracer.activeSpanSource()`

Reply:
`tracer.activeSpanSource` isn't public. Further, it is defined in `com.uber.jaeger.Tracer` in `jaeger-core`, for which we'd have to add a dependency from `jaeger-context` instead of depending on `opentracing-api`.